### ISSUE TITLE
Parsing XML with namespaces

### ISSUE THREAD

**Issue by ajsait** [drmohundro/SWXMLHash#30]
Currently, there is no way to set the XMLParser shouldProcessNamespaces to true. So when accessing an element, the complete tagname with namespace needs to be passed. e.g xml["f:animal"] instead of just xml["animal"]

In the example in the code, it would be nice if instead of this

**xml["root"]["h:table"]["h:tr"]["h:td"][0].element!.text!**

we will have this

**xml["root"]["table"]["tr"]["td"][0].element!.text!**

**Comment by drmohundro** [comment]
Thanks for the suggestion - I like the idea and it should be relatively easy to support.

Just thinking out loud, but my initial thought is to add an optional options parameter to `parse` to allow for setting things like this. Any thoughts on that?

**Comment by ajsait** [comment]
I have no problem with that. I think that would also be my approach.

**Comment by drmohundro** [comment]
Sorry for the delayed response - I didn't even see your question regarding how long it would take. It honestly should be a pretty quick change.

I actually had another thought regarding a variation in the API - I thought a fluent API where a config parameter was passed to an action might make a little more sense, like so:

```swift
SWXMLHash.configure {
  config in
  config.shouldProcessNamespaces = true
}.parse(xml)
```

That way, it would be easy to add additional configuration options to set ahead of time as opposed to adding multiple parameters to the parse method. As an aside, it might be a better way to handle the lazy parsing, too.

Quick question for you - are you currently targeting the new Swift 2.0 in beta or are you still on Swift 1.2 (currently released)? I've set up a branch for 2.0 and can make the changes there, but if this would help you on 1.2, I can make the change there and then merge it over.

**Comment by ajsait** [comment]
We are still using swift 1.2

**Comment by drmohundro** [comment]
I've created #34 to track this work - there are tests that pass. The general usage is:

```
parser = SWXMLHash.config { conf in
    conf.shouldProcessNamespaces = true
}.parse(xmlWithNamespace)
```

I'd like to keep the pull request open for a little while to see if anyone has any additional input on using the `config` method like that. I've used that approach in the past for my own projects and a few other libraries, but I haven't seen it used much in any Swift projects yet. If I don't hear any feedback, though, I'll go ahead and merge it into master.
